runtime/src/main/java/org/corfudb/runtime/view/AbstractReplicationView.java

```diff
@@ -40,20 +40,7 @@ public AbstractReplicationView(Layout layout, Layout.LayoutSegment ls) {
     public static AbstractReplicationView getReplicationView(Layout l, Layout.ReplicationMode mode, Layout.LayoutSegment ls) {
         if (l.replicationViewCache == null) { l.replicationViewCache = new ConcurrentHashMap<>(); } //super hacky
         return l.replicationViewCache.computeIfAbsent(ls, x -> {
-            // TODO: really broken software engineering here... refactor!
-            switch (ls.getReplicationMode()) {
-                case CHAIN_REPLICATION:
-                    return new ChainReplicationView(l, ls);
-                case QUORUM_REPLICATION:
-                    log.warn("Quorum replication is not yet supported!");
-                    break;
-                case REPLEX:
-                    return new ReplexReplicationView(l, ls);
-                default:
-                    log.error("Unknown replication mode {} selected.", mode);
-                    break;
-            }
-            return null;
+            return mode.getReplicationView(l, ls);
```

Comment:
In theory, you can change the replication mode with a layout change, and every segment in the global sequence of operations is independent from others.

In practice, I don't envision this being terribly useful.